src/foam/dao/FixedSizeDAO.js

```diff
@@ -82,7 +93,12 @@ foam.CLASS({
   } finally {
     getLock().unlock();
   }
+  foam.core.FObject discard = getFixedSizeArray()[insertAt];
```

Comment:
It seems really wasteful of memory that we're storing the entire object in an array in memory as well as potentially storing the objects in the delegate MDAO. Unrelated to your changes here, but maybe we should rethink that. Maybe we could just be storing the ids?

Reply:
This would suggest a FixedSizeMDAO - if we are storing in memory. 
The FixedSizeDAO must know if the delegate DAO is actually storing, consider LoggingDAO and which is setup with an NullDAO rather than an MDAO.